graph/src/components/store.rs

```diff
@@ -821,17 +824,13 @@ pub struct StoredDynamicDataSource {
     pub creation_block: Option<BlockNumber>,
 }
 
-impl From<&DataSource> for StoredDynamicDataSource {
-    fn from(ds: &DataSource) -> Self {
+impl<DS: DataSource> From<&DS> for StoredDynamicDataSource {
+    fn from(ds: &DS) -> Self {
```

Comment:
Longer term, we want the specific blockchain to provide this conversion, and probably need to put the appropriate constraint on `blockchain::DataSource`. It's fine for now, but maybe worth an `ETHDEP` comment

Reply:
Giving blockchains flexibility here is about more than just the conversion but also about the fields of `StoredDynamicDataSource`. `name`, `context` and `creation_block` can all be common, but what to do with the `Source` type is an open question. The solution here will fall out of how we solve the `Source` type.